extensions/appdynamics/extension.py

```diff
@@ -0,0 +1,202 @@
+            self._log.debug("Log Path [%s]", self.log_path)
+            self.daemon_path = os.path.join(
+                '@{HOME}', 'appdynamics', 'daemon',
+                'appdynamics-daemon.%s' % self._php_arch)
+            self._log.debug("Daemon [%s]", self.daemon_path)
+            self.socket_path = os.path.join('@{HOME}', 'appdynamics',
+                                            'daemon.sock')
+            self._log.debug("Socket [%s]", self.socket_path)
+            self.pid_path = os.path.join('@{HOME}', 'appdynamics',
+                                         'daemon.pid')
+            self._log.debug("Pid File [%s]", self.pid_path)
+
+    def _load_php_info(self):
+        self.php_ini_path = os.path.join(self._ctx['BUILD_DIR'],
+                                         'php', 'etc', 'php.ini')
+        self._php_extn_dir = self._find_php_extn_dir()
+        self._php_api, self._php_zts = self._parse_php_api()
+        self._php_arch = self._ctx.get('APPDYNAMICS_ARCH', 'x64')
+        self._log.debug("PHP API [%s] Arch [%s]",
+                        self._php_api, self._php_arch)
+
+    def _find_php_extn_dir(self):
+        with open(self.php_ini_path, 'rt') as php_ini:
+            for line in php_ini.readlines():
+                if line.startswith('extension_dir'):
+                    (key, val) = line.strip().split(' = ')
+                    return val.strip('"')
+
+    def _parse_php_api(self):
+        tmp = os.path.basename(self._php_extn_dir)
+        php_api = tmp.split('-')[-1]
+        php_zts = (tmp.find('non-zts') == -1)
+        return php_api, php_zts
+
+    def should_install(self):
+        return self._detected
+
+# Extension Methods
+def preprocess_commands(ctx):
+
+    service = ctx.get('VCAP_SERVICES', {})
+    service_defs = service.get('appdynamics', [])
+    detected = False
+    if len(service_defs) == 0:
+       _log.info("AppDynamics services with tag appdynamics not detected.")
+       _log.info("Looking for tag app-dynamics service.")
+       service_defs = service.get('app-dynamics', [])
+       if len(service_defs) == 0:
+          _log.info("AppDynamics services with tag app-dynamics not detected.")
+          _log.info("Looking for Appdynamics user-provided service.")
+          cups_service_defs = service.get('user-provided', [])
+
+          if len(cups_service_defs) == 0:
+             _log.info("AppDynamics services not detected.")
+          else:
+             cups_svc = cups_service_defs.get('name', [])
+             if (cups_svc == "appdynamics") or (cups_svc == "app-dynamics"):
+                _log.info("AppDynamics cups services detected.")
+                detected = True
+
+    if len(service_defs) > 0:
+        _log.debug("AppDynamics service detected.")
+        detected = True
+
+    if detected == True: 
+    	exit_code = os.system("echo preprocess_commands: AppDynamics agent configuration")
+        return [[ 'echo', '" in preprocess;"'],
+                ['env'],
```

Comment:
This is going to dump the full environment, which could expose sensitive info like passwords.  Remove this unless there's a very good reason.